pandas/core/interchange/from_dataframe.py

```diff
@@ -181,9 +180,15 @@ def categorical_column_to_series(col: Column) -> tuple[pd.Series, Any]:
         raise NotImplementedError("Non-dictionary categoricals not supported yet")
 
     cat_column = categorical["categories"]
-    # for mypy/pyright
-    assert isinstance(cat_column, PandasColumn), "categories must be a PandasColumn"
-    categories = np.array(cat_column._col)
+    if hasattr(cat_column, "_col"):
```

Comment:
What is the reason for using this attribute? (that also seems like relying on and exposing some internal detail) 
Haven't looked in detail, but my understanding is that this object is just a protocol Column object, and we already have all the code to convert that to a pandas object?

Reply:
I don't know, this was already there - looks like it was introduced in https://github.com/pandas-dev/pandas/pull/47886/files

Reply:
Yeah, I know it was already there, but IMO the "proper" fix for the reported issue is to stop relying on this detail. 
Because the the test you added only "happens" to work because pyarrow's Column object also uses `_col` to store the original array; if we would rename that internal property (and I would say it's actually a confusing name in hindsight, since it's an array ;)), the test would start failing.

Reply:
yup, agreed